packages/front-end/components/Features/ScheduleInputs.tsx

```diff
@@ -3,50 +3,77 @@ import clsx from "clsx";
 import { format } from "date-fns";
 import { format as formatTimeZone } from "date-fns-tz";
 import React, { useEffect, useState } from "react";
+import { useUser } from "@/services/UserContext";
 import Field from "../Forms/Field";
 import SelectField from "../Forms/SelectField";
+import PremiumTooltip from "../Marketing/PremiumTooltip";
+import UpgradeMessage from "../Marketing/UpgradeMessage";
+import Toggle from "../Forms/Toggle";
 import styles from "./ScheduleInputs.module.scss";
 
 interface Props {
   defaultValue: ScheduleRule[];
   onChange: (value: ScheduleRule[]) => void;
+  setShowUpgradeModal: (value: boolean) => void;
 }
 
 export default function ScheduleInputs(props: Props) {
   const [rules, setRules] = useState(props.defaultValue);
   const [dateErrors, setDateErrors] = useState({});
+  const { hasCommercialFeature } = useUser();
+  const [toggleEnabled, setToggleEnabled] = useState(() => rules?.length > 0);
+
+  const canScheduleFeatureFlags = hasCommercialFeature("schedule-feature-flag");
 
   useEffect(() => {
     props.onChange(rules);
   }, [props, props.defaultValue, rules]);
 
+  useEffect(() => {
+    if (toggleEnabled && !rules.length) {
+      setRules([
+        {
+          enabled: true,
+          timestamp: null,
+        },
+        {
+          enabled: false,
+          timestamp: null,
+        },
+      ]);
+    } else if (toggleEnabled && rules.length) {
+      setRules(rules);
+    } else {
+      setRules([]);
```

Comment:
Does this clear out the rules when you disable the toggle? It might be good to avoid clearing them out in case the user toggles it off and back on again. You might just check the toggle state during submit to include/exclude the rules from the payload.

Reply:
It does. I've gone back and forth on this myself. I need to figure out a way to not clear the rules, and pass the state of `toggleEnabled` into the parent component, so if `toggleEnabled` is false, we ignore the rules when submitting the form.

Reply:
One possible route is, we add a property that is solely responsible for the state of the toggle, so we can decouple the `scheduleRules` from the toggle.

This would allow us to persist any scheduleRules even if the user disables the toggle and infer the initial state of the toggle from that property. 

I think I would do it in such a way that if the user submitted the form with the toggle disabled, the scheduleRules would be wiped.

Reply:
@bryce-fitzsimons Just pushed up changes based on the comment above - let me know what you think!

Reply:
@bryce-fitzsimons I really appreciate the time/effort you've put into this PR review. Just pushed up some changes based on your latest round of feedback.